src/request.js

```diff
@@ -82,11 +100,23 @@ export default class Request {
 			}
 		}
 
+		const abortController = new AbortController();
+		if (signal !== undefined) {
+			if (signal.aborted) {
+				abortController.abort();
+			} else {
+				signal.addEventListener('abort', () => {
```

Comment:
This eventListener is never removed, is it?

Reply:
If it could be of any help maybe adding {once: true} to addEventListener could help cleaning up things

Reply:
@mika-fischer No, and per spec we can't, so we'll just have to rely on GC to do its job. We could add a `{ once: true }` though.

Reply:
But won't this cause issues, for instance in the case where I have a long running Node.js application where I have a global AbortController (to be able to abort all requests when the application shuts down) and each request in the whole lifetime of the app (let's say hundreds of thousands) adds a listener, none of which is ever removed?

Reply:
😔 yes. It’s also one of the reasons why this is a WIP.

Reply:
Can't you just save a cleanup closure (which just removes the listener from the parent AbortController) and attach this closure to the appropriate events of the Node.js Request/Response so that it gets called when the request is finished?

Reply:
A cleanup closure would remove the listener from the parent AbortController after fetch() terminates for any reason, right? Would that affect the ability to cancel a fetch that reuses a Request object for multiple fetches? Does that maybe mean that the listener should be added each time the fetch begins and removed when the fetch terminates?

Reply:
`.once()` seems like it would only help if the request terminated by being aborted, but I might be missing something.